### DISCUSSION TITLE
Follow-up on Diagram Creation in Tree Description

### DISCUSSION THREAD

**Discussion by hoajie** [eclipse-sirius/sirius-desktop#484] [Q&A] [answered]
Dear Sirius team,

I hope this message finds you well. Yesterday, I posted a question regarding how to use Diagram Creation in a Tree Description, and I greatly appreciate Laurent Redor's prompt response. Following his advice, I was able to successfully implement Diagram Creation, so thank you for that!

I now have a follow-up question regarding an issue I've encountered while using Diagram Creation in the tree structure. Here's what I am trying to achieve:

I have created two Diagram Creations in the tree, named Diagram Creation on Package1 and Diagram Creation on Package2. I also created two different Diagram Descriptions, named Diag on Package 1 and Diag on Package 2. In Diagram Creation on Package1, I used the Mappings and Diagram Description to select Diag on Package 1; similarly, I selected Diag on Package 2 for Diagram Creation on Package2.

My expectation is that when I click on a package in the new tree, I should be able to choose different Diagram Creations to create different Diagram Descriptions. For example, clicking on Diagram Creation on Package1 should create Diag on Package 1, and clicking on Diagram Creation on Package2 should create Diag on Package 2. Does this approach seem correct to you?

Additionally, I have one more question regarding the UI. When we launch the action "New/Diagram Creation on Package" by right-clicking on a package, the context menu displays the label "New". I am wondering where this "New" label is defined, as I would like to customize it (e.g., change "New" to "New Diagram"). Could you please advise how I can modify this label?

Thank you for your support, and I look forward to your guidance on these issues!

Best regards,
![Snipaste_2024-10-23_13-45-27](https://github.com/user-attachments/assets/48aaed68-6a01-44c7-b6b4-da863fa6e571)
![Snipaste_2024-10-23_13-45-41](https://github.com/user-attachments/assets/2d9a338d-0b97-4bb8-b82c-2c97d3807a76)
![Snipaste_2024-10-23_13-45-56](https://github.com/user-attachments/assets/a0fd1f32-5bdb-4337-93ad-b59c5f69ff50)
![Snipaste_2024-10-23_13-46-09](https://github.com/user-attachments/assets/5deed99f-0dae-449c-b1d4-5d695fb1f3f9)

**Comment by lredor** [comment, marked as answer]
Hi,
`Does this approach seem correct to you?` --> Yes, it is, but I'm not sure, what is the real question. I complete the sample of the previous discussion with two diagrams (one with blue classes and the other one with red classes) and two diagram creation menus and so with the expected behavior: [TreeSampleProject-v2.zip](https://github.com/user-attachments/files/17486492/TreeSampleProject-v2.zip).
![2024-10-23_08h55_19](https://github.com/user-attachments/assets/2ccce929-3dcf-4241-a349-c64f7b5f0b20)

The "New" menu in the contextual menu is "hard-coded" [here](https://github.com/eclipse-sirius/sirius-desktop/blob/master/plugins/org.eclipse.sirius.tree.ui/src/org/eclipse/sirius/tree/ui/tools/internal/editor/DTreeMenuListener.java#L220), so you can not customize it. Indeed, this menu can contain diagram creation tools, but also tree or table creation tools.
Regards,
Laurent

**Reply by hoajie** [reply]
Hi Laurent,

Thank you so much for your prompt response and for providing the updated sample project. Your reply has been incredibly helpful!

Yes, you're right—I was aiming to create different Diagram Descriptions, much like in your example with the blue and red classes for displaying ecore files. This is exactly the behavior I was looking to achieve, so I greatly appreciate your assistance!

Just to confirm, you mentioned that the "New" label in the context menu (when we right-click and launch the action "New/Diagram Creation on Package") is hard-coded and cannot be modified, is that correct? If that's the case, I'll simply proceed without customizing it and will use the default "New" as per your suggestion.

Once again, thank you for your support!

Best regards,
hoajie

**Reply by lredor** [reply]
`Is that correct?` --> Yes.
